File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.java
import java.util.*;
class Solution {
    public Stack<Integer> solution(int[] arr) {
        Stack<Integer> st = new Stack<>();
        
        for(int i=0; i < arr.length; i++){
            if(st.isEmpty()) st.push(arr[i]);
            else if(!st.isEmpty() && st.peek() == arr[i]) st.pop();
            else if(!st.isEmpty() && st.peek() != arr[i]) st.push(arr[i]);
        }
        
        if(st.isEmpty()) st.push(-1);
        return st;
    }
}

File: 프로그래머스/0/181859. 배열 만들기 6/배열 만들기 6.kt
import java.util.Stack

class Solution {
    fun solution(arr: IntArray): IntArray {
        val st = Stack<Int>()
        var i = 0
        while(i < arr.size){
            if(st.isEmpty() || st.peek() != arr[i]){
                st.push(arr[i])
                i++
            }else if(st.peek() == arr[i]){
                st.pop()
                i++
            }
        }
        return if(st.isEmpty()) intArrayOf(-1) else st.toIntArray()
    }
}